daemon/container_operations.go

```diff
@@ -1084,3 +1129,31 @@ func (daemon *Daemon) DeactivateContainerServiceBinding(containerName string) er
 	}
 	return sb.DisableService()
 }
+
+// isNetworkLazyCreated checks if the input network is currently not found on this worker node,
+// but after sometime it will be gossiped to this node as lazy creation.
+func (daemon *Daemon) isNetworkLazyCreated(idOrName string) (libnetwork.Network, bool, error) {
+	// if daemon is not a part of swarm, just return false
+	if !daemon.cluster.IsManager() && !daemon.cluster.IsAgent() {
+		return nil, false, nil
+	}
+
+	for i := 0; i < 3; i++ {
+		n, err := daemon.FindNetwork(idOrName)
+		if err != nil {
+			if _, ok := err.(libnetwork.ErrNoSuchNetwork); !ok {
+				// if the error is not NotFound, just return err
+				return nil, false, err
+			}
+
+			// network is not found, continue the loop
+			time.Sleep(1 * time.Second)
```

Comment:
I am afraid this is improper to hard code the duration. I think this is your concern. While this design needs more review from you maintainers.